[CI] OpenAiServiceUpgradeIT testOpenAiEmbeddings {upgradedNodes=1} failing

**Build Scans:**
- [elasticsearch-periodic #5562 / 8.12.2_bwc](https://gradle-enterprise.elastic.co/s/s67unss7qf3ba)
- [elasticsearch-periodic #5557 / 8.12.2_bwc](https://gradle-enterprise.elastic.co/s/ahgkaqdo6za7s)
- [elasticsearch-periodic #5552 / 8.12.2_bwc](https://gradle-enterprise.elastic.co/s/tvo5hqibgfff6)
- [elasticsearch-periodic #5547 / 8.12.2_bwc](https://gradle-enterprise.elastic.co/s/3442x7fbn6t5w)
- [elasticsearch-periodic #5542 / 8.12.2_bwc](https://gradle-enterprise.elastic.co/s/dpucu2kubvwqi)
- [elasticsearch-periodic #5537 / 8.12.2_bwc](https://gradle-enterprise.elastic.co/s/fg657jqepnk6y)
- [elasticsearch-periodic #5532 / 8.12.2_bwc](https://gradle-enterprise.elastic.co/s/chyzhi7o7626s)
- [elasticsearch-periodic #5527 / 8.12.2_bwc](https://gradle-enterprise.elastic.co/s/rk473yzqjvsos)
- [elasticsearch-periodic #5522 / 8.12.2_bwc](https://gradle-enterprise.elastic.co/s/6y3u4woq4pav6)
- [elasticsearch-periodic #5517 / 8.12.2_bwc](https://gradle-enterprise.elastic.co/s/ybndrwhapqa6i)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:inference:qa:rolling-upgrade:v8.12.2#bwcTest" -Dtests.class="org.elasticsearch.xpack.application.OpenAiServiceUpgradeIT" -Dtests.method="testOpenAiEmbeddings {upgradedNodes=1}" -Dtests.seed=23497966C12E4985 -Dtests.bwc=true -Dtests.locale=os-RU -Dtests.timezone=Australia/Lindeman -Druntime.java=23
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.application.OpenAiServiceUpgradeIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testOpenAiEmbeddings%20%7BupgradedNodes%3D1%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: model_id not found in config: [{service_settings={similarity=dot_product, url=http://127.0.0.1:39131, dimensions=2}, service=openai, task_settings={model=text-embedding-ada-002}, model_id=old-cluster-embeddings, task_type=text_embedding}]
```

**Issue Reasons:**
- [8.x] 21 consecutive failures in step 8.12.2_bwc
- [8.x] 21 failures in test testOpenAiEmbeddings {upgradedNodes=1} (2.9% fail rate in 735 executions)
- [8.x] 21 failures in step 8.12.2_bwc (100.0% fail rate in 21 executions)
- [8.x] 21 failures in pipeline elasticsearch-periodic (100.0% fail rate in 21 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/inference/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/xpack/application/OpenAiServiceUpgradeIT.java
+++ b/x-pack/plugin/inference/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/xpack/application/OpenAiServiceUpgradeIT.java
@@ -85,8 +85,9 @@ public class OpenAiServiceUpgradeIT extends InferenceUpgradeTestCase {
             assertEquals("openai", configs.get(0).get("service"));
             var serviceSettings = (Map<String, Object>) configs.get(0).get("service_settings");
             var taskSettings = (Map<String, Object>) configs.get(0).get("task_settings");
-            var modelIdFound = serviceSettings.containsKey("model_id") || taskSettings.containsKey("model_id");
-            assertTrue("model_id not found in config: " + configs.toString(), modelIdFound);
+            var modelIdFound = serviceSettings.containsKey("model_id") || (taskSettings.containsKey("model") &&
+                getOldClusterTestVersion().onOrBefore(OPEN_AI_EMBEDDINGS_MODEL_SETTING_MOVED));
+            assertTrue("model_id not found in config: " + configs, modelIdFound);
 
             assertEmbeddingInference(oldClusterId);
         } else if (isUpgradedCluster()) {

--- a/x-pack/plugin/inference/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/xpack/application/OpenAiServiceUpgradeIT.java
+++ b/x-pack/plugin/inference/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/xpack/application/OpenAiServiceUpgradeIT.java
@@ -85,8 +85,8 @@ public class OpenAiServiceUpgradeIT extends InferenceUpgradeTestCase {
             assertEquals("openai", configs.get(0).get("service"));
             var serviceSettings = (Map<String, Object>) configs.get(0).get("service_settings");
             var taskSettings = (Map<String, Object>) configs.get(0).get("task_settings");
-            var modelIdFound = serviceSettings.containsKey("model_id") || (taskSettings.containsKey("model") &&
-                getOldClusterTestVersion().onOrBefore(OPEN_AI_EMBEDDINGS_MODEL_SETTING_MOVED));
+            var modelIdFound = serviceSettings.containsKey("model_id")
+                || (taskSettings.containsKey("model") && getOldClusterTestVersion().onOrBefore(OPEN_AI_EMBEDDINGS_MODEL_SETTING_MOVED));
             assertTrue("model_id not found in config: " + configs, modelIdFound);
 
             assertEmbeddingInference(oldClusterId);

--- a/x-pack/plugin/inference/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/xpack/application/OpenAiServiceUpgradeIT.java
+++ b/x-pack/plugin/inference/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/xpack/application/OpenAiServiceUpgradeIT.java
@@ -16,6 +16,8 @@ import org.junit.AfterClass;
 import org.junit.BeforeClass;
 
 import java.io.IOException;
+import java.util.Collection;
+import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
 
@@ -139,9 +141,11 @@ public class OpenAiServiceUpgradeIT extends InferenceUpgradeTestCase {
 
             assertCompletionInference(oldClusterId);
         } else if (isMixedCluster()) {
-            var configs = (List<Map<String, Object>>) get(testTaskType, oldClusterId).get("endpoints");
+            List<Map<String, Object>> configs = new LinkedList<>();
+            var request = get(testTaskType, oldClusterId);
+            configs.addAll((Collection<? extends Map<String, Object>>) request.getOrDefault("endpoints", List.of()));
             if (oldClusterHasFeature("gte_v" + MODELS_RENAMED_TO_ENDPOINTS) == false) {
-                configs.addAll((List<Map<String, Object>>) get(testTaskType, oldClusterId).get(old_cluster_endpoint_identifier));
+                configs.addAll((List<Map<String, Object>>) request.getOrDefault(old_cluster_endpoint_identifier, List.of()));
                 // in version 8.15, there was a breaking change where "models" was renamed to "endpoints"
             }
             assertEquals("openai", configs.get(0).get("service"));

